langroid/agent/special/sql_chat_agent.py

```diff
@@ -288,3 +292,75 @@ def extract_and_combine_db_info(
                     )
 
     return db_info
+
+
+def extract_schema_descriptions(engine: Engine) -> Dict[str, Dict[str, Any]]:
+    """
+    Extracts the schema descriptions from the database connected to by the engine.
+
+    Args:
+        engine (Engine): SQLAlchemy engine instance.
+
+    Returns:
+        Dict[str, Dict[str, Any]]: A dictionary representation of table and column
+        descriptions.
+    """
+    dialect_name = engine.dialect.name
+    inspector = inspect(engine)
+    table_names: List[str] = inspector.get_table_names()
+
+    result: Dict[str, Dict[str, Any]] = {}
+
+    # PostgreSQL
+    if dialect_name == "postgresql":
```

Comment:
As we add more dialects, this fn will get long and duplicative. Suggest refactoring this to handle each dialect in a separate fn, or maybe have separate classes if that makes sense.

Reply:
Will do, there are only 2 other dialects supported by SQLAlchemy that support built-in descriptions. So separate fn should be enough.